controllers/lock_client.go

```diff
@@ -233,6 +248,79 @@ func (client *RealLockClient) ClearPendingUpgrades() error {
 	return err
 }
 
+// GetDenyList retrieves the current deny list from the database.
+func (client *RealLockClient) GetDenyList() ([]string, error) {
+	list, err := client.database.Transact(func(tr fdb.Transaction) (interface{}, error) {
+		err := tr.Options().SetReadSystemKeys()
+		if err != nil {
+			return nil, err
+		}
+
+		keyRange, err := client.getDenyListKeyRange()
+		if err != nil {
+			return nil, err
+		}
+
+		values := tr.GetRange(keyRange, fdb.RangeOptions{}).GetSliceOrPanic()
+		list := make([]string, len(values))
+		for index, value := range values {
+			list[index] = string(value.Value)
+		}
+		return list, nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return list.([]string), nil
```

Comment:
We may want to check if the cast is possible to prevent a fatal error?

Reply:
I'd rather avoid checks for things that we believe to be impossible, to reduce code clutter. Given that we know that `err` is nil at this point, we should be confident that `list` is a valid slice.